src/Riok.Mapperly/Descriptors/MappingBuilders/DictionaryMappingBuilder.cs

```diff
@@ -135,6 +138,34 @@ private static (INewInstanceMapping, INewInstanceMapping)? BuildKeyValueMapping(
         return (keyMapping, valueMapping);
     }
 
+    private static INewInstanceMapping? TryGetFromEnumerable(
+        MappingBuilderContext ctx,
+        INewInstanceMapping keyMapping,
+        INewInstanceMapping valueMapping
+    )
+    {
+        if (!keyMapping.IsSynthetic || !valueMapping.IsSynthetic || keyMapping.TargetType.IsNullable())
+            return null;
+
+        // use .NET 8  method .ToDictionary() if value and key mapping is synthetic
+        var enumerableType = ctx.Types.Get(typeof(IEnumerable<>));
+        var dictionaryType = ctx.Types.Get(typeof(Dictionary<,>));
+
+        var fromEnumerableCtor = dictionaryType.Constructors.FirstOrDefault(
+            x =>
+                x is { Parameters.Length: 1 }
```

Comment:
This code base usually uses pattern matching only for multiple conditions and/or type checks. Probably use `x.Parameters.Length == 1` here.